pkgs/development/libraries/opencv/4.x.nix

```diff
@@ -227,6 +229,22 @@ let
   #multithreaded openblas conflicts with opencv multithreading, which manifest itself in hung tests
   #https://github.com/xianyi/OpenBLAS/wiki/Faq/4bded95e8dc8aadc70ce65267d1093ca7bdefc4c#multi-threaded
   openblas_ = blas.provider.override { singleThreaded = true; };
+
+  inherit (cudaPackages) cudaFlags cudatoolkit cudaVersion;
+  inherit (cudaFlags) cudaCapabilities;
+
+  cuda-redist = symlinkJoin {
+    name = "cuda-redist-${cudaVersion}";
+    paths = with cudaPackages; ([
+      cuda_cccl # <thrust/*>
+      cuda_cudart # cuda_runtime.h
+      cuda_nvcc
+      libnpp
+    ] ++ lib.optional enableCublas libcublas
+      ++ lib.optional enableCudnn cudnn
+      ++ lib.optional enableCufft libcufft
+    );
+   };
```

Comment:
@ConnorBaker break this into the `cuda-native-redist`/`cuda-redist` pattern you've used previously.